scripts/zones/Southern_San_dOria/npcs/Deraquien.lua

```diff
@@ -30,11 +31,49 @@ end;
 
 function onTrigger(player,npc)
 
+    local Fame = player:getFameLevel(SANDORIA);
+    local Rank = player:getRank();
+    local Nation = player:getNation();
+    -- access to chateau if not sandorian
+    local Access = (player:getCurrentMission(Nation) == 5) or (player:hasCompletedMission(Nation, 5));
```

Comment:
technically, San d'Orians should be checked here as well, the only difference being you check mission 1-3 (where you have access to the Chateau) instead of checking rank

Reply:
What would be nice is a `player:hasChateauAccess()`
I think it could be used more than just in this quest.